python/cuspatial/cuspatial/core/spatial/join.py

```diff
@@ -238,7 +239,8 @@ def quadtree_point_in_polygon(
         polygon_index : cudf.Series
             Indices of each polygon with which a point intersected.
         point_index : cudf.Series
-            Indices of each point that intersects with a polygon.
+            Indices of each point (in ``point_indices``) that intersects with a
```

Comment:
```suggestion
            Indices of points (in ``point_indices``) that are contained in a
```

Reply:
Also, I find the "(in ``point indices``) a bit unclear.  Are the actual indices of a contained polygon returned, or are the indices of the points in `point_indices` returned? i.e. is it this:

```
point_indices = [0, 5, 9, 21, 45]
contained_points = [0, 9, 45]
return contained_points
```

or this:

```
point_indices = [0, 5, 9, 21, 45]
contained_points = [0, 2, 4]
return contained_points
```

Reply:
The second one. The returned indices can be used as a gather map for `point_indices`, which is itself a gather map for `point_x` and `point_y`:

```python
points = cudf.DataFrame({"x": [-5, 5], "y": [-5, 5]})

indices, quadtree = cuspatial.quadtree_on_points(points["x"], points["y"], ...)
bboxes = cuspatial.polygon_bounding_boxes(...)
intersections = cuspatial.join_quadtree_and_bounding_boxes(quadtree, bboxes, ...)
polygons_and_points = cuspatial.quadtree_point_in_polygon(
  intersections, quadtree, indices, points["x"], points["y"], ...
)

intersecting_points = points[indices[polygons_and_points["point_index"]]]
```

Reply:
OK, so these are indices of indices. That is very hard to explain succinctly. I think an example that makes this clear should be added to the docstring.  And I would word it something like:

```suggestion
            Index of contained point. This index refers to ``point_indices``, so it is an index to an index.
```

Note I suggested changing the working of the polygon index above to match. Is the polygon index an index to an index as well?